docs/hub/tutorial-add-library.md

```diff
@@ -0,0 +1,184 @@
+```
+
+Download a specific version of the file by specifying the `revision` parameter. The `revision` parameter can be a branch name, tag, or commit hash. 
+
+The commit hash must be a full-length hash instead of the shorter 7-character commit hash:
+
+```python
+>>> from huggingface_hub import hf_hub_download
+>>> hf_hub_download(repo_id="lysandre/arxiv-nlp", filename="config.json", revision="877b84a8f93f2d619faa2a6e514a32beef88ab0a")
+```
+
+Use the `cache_dir` parameter to change where a file is stored:
+
+```python
+>>> from huggingface_hub import hf_hub_download
+>>> hf_hub_download(repo_id="lysandre/arxiv-nlp", filename="config.json", cache_dir="/home/lysandre/test")
+```
+
+### Code sample
+
+We recommend adding a code snippet to explain how a model should be used in your downstream library. 
+
+![/docs/assets/hub/code_snippet.png](/docs/assets/hub/code_snippet.png)
+
+Add a code snippet by updating the [Libraries Typescript file](https://github.com/huggingface/huggingface_hub/blob/main/widgets/src/lib/interfaces/Libraries.ts) with instructions for your model. For example, the spaCy integration includes a brief code snippet for how to load and use a spaCy model:
+
+```typescript
+const spacy = (model: ModelData) =>
+`!pip install https://huggingface.co/${model.modelId}/resolve/main/${nameWithoutNamespace(model.modelId)}-any-py3-none-any.whl
+# Using spacy.load().
+import spacy
+nlp = spacy.load("${nameWithoutNamespace(model.modelId)}")
+# Importing as module.
+import ${nameWithoutNamespace(model.modelId)}
+nlp = ${nameWithoutNamespace(model.modelId)}.load()`;
+```
+
+This will also add a tag to your model so users can quickly identify models from your library.
+
+![/docs/assets/hub/libraries-tags.png](/docs/assets/hub/libraries-tags.png)
+
+## Upload files to the Hub
+
+You will also want to provide a method for creating model repositories and uploading files to the Hub directly from your library. The `HfApi` class has two methods to assist you with creating repositories and uploading files:
+
+- `create_repo` creates a repository on the Hub.
+- `upload_file` directly uploads files to a repository on the Hub.
+
+Begin by instantiating the `HfApi` class:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+```
+
+You will also need to retrieve your Hugging Face API token to create a repository and upload files to it. Retrieve your token with the `HfFolder` method:
+
+```python
+>>> from huggingface_hub import HfFolder
+>>> token = HfFolder().get_token()
+```
+
+### `create_repo`
+
+The `create_repo` method creates a repository on the Hub. Use the `name` parameter to provide a name for your repository:
+
+```python
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+>>> api.create_repo(token=token, name="test-model")
+'https://huggingface.co/lysandre/test-model'
+```
+
+When you check your Hugging Face account, you should now see a `test-model` repository under your namespace.
+
+### `upload_file`
```

Comment:
This approach is fine if you want to upload a single file, but many times you might want to upload many files. Maybe it's worth mentioning you can do this and link to the `Repository` documentation. WDYT?

Reply:
Good idea, added link to `Repository` documentation